records/2024-05-23.scalability.short-code-implementation.md

```diff
@@ -0,0 +1,107 @@
+| Verification message | Authentication codes <br>Confirmation of account changes |
+| Confirmation / Receipts | Record of transactions<br>Approvals |
+| Other | User defined |
+
+### Determining Template Priority
+A new table will be added to the DB for TemplateCategories. The primary key from TemplateCategories will be a foreign key to the Templates Table. All TemplateCategories will be assigned a pre-computed priority.
+
+<!-- ![Potential ERD](https://raw.githubusercontent.com/cds-snc/notification-adr/records/diagrams/2024-05-23.scalability.short-code-implementation/ERD.jpg?raw=true "Potential ERD") -->
+
+![Potential ERD](../records/diagrams/2024-05-23.scalability.short-code-implementation/ERD.jpg?raw=true "Potential ERD")
+
+```
+process_type is (bulk, normal, priority)
+```
+
+Given the new TemplateCategories, each template will default to a process_type that is set in TemplateCategory. To create manual overrides, an Admin will edit the process_type on the Templates Table. For this to work, we have two options:
+1. Create a new column in Templates `new_process_type`
+2. Clear the current process_type in the Templates table to be empty.
+
+After the above is implemented,
+```
+Notification Priority = Template.new_process_type else TemplateCategory.process_type
+```
+For most Templates, the Template.process_type/new_process_type will be empty. Only if overridden by Admin, will that field have information.
+
+**Use case:**  
+An SMS Template is of type "XYZ" that is of "low" priority. The client reaches out to us about the importance of the messages being sent immediately. We leave the category of the Template as "XYZ", but manually update the Template to "high" priority. This means that the message will be sent using the short-code.
+
+
+### Other Template Category
+
+Option 1:  
+Let client's manually add a Category for their template, if it doesn't fit in the above usecase. A ticket is created in Freshdesk for this new Template Category. A decision is made about adding a new Category, support can add that Category through an Admin workflow page (similar to Branding).
+
+Option 2:  
+Client adds a new Category. We run a classifier to the Templates and determine whether it should be a new Category, or live within the existing Categories (supervised learning). Depending on the decision, a new entry is made in the TemplateCategory table.
+
+Regardless of which option we choose, there is a time when the Category for a Template is undetermined. During this time, we will default the process_type of that Template to a Generic "Low Category (Bulk)" for all of our workflows to stay the same.
+
+#### Example
+There is a process_type in TemplateCategories and process_type in Templates.
+
+1. If a client chooses a category, the process_type in Templates will be empty.
+1. If the client chooses "Other" - while we are determining what TemplateCategory the Template must have, it will default to "Low".
```

Comment:
Which table does the process type default to *Low*? Do we mean the `TemplateCategory` one here?

Reply:
correct, we will have a TemplateCategory "Default Low" , and "Other" can be attached to that Category till we decide what the category is